tooling/typescript/base.json

```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "compilerOptions": {
+    /** Base Options */
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "target": "ES2022",
+    "lib": ["ES2022"],
+    "allowJs": true,
+    "resolveJsonModule": true,
+    "moduleDetection": "force",
+    "isolatedModules": true,
+
+    /** Keep TSC performant in monorepos */
+    "incremental": true,
+    "disableSourceOfProjectReferenceRedirect": true,
+
+    /** Strictness */
+    "strict": true,
+    "noUncheckedIndexedAccess": true,
+    "checkJs": true,
+
+    /** Transpile using Bundler (not tsc) */
+    "module": "ESNext",
+    "moduleResolution": "Bundler",
+    "noEmit": true
```

Comment:
i'm assuming this defaults back to whatever bundler we're using? i'm not too sure how different is this vs `tsc` though - is the difference just that using bundler emits a single file?

Reply:
this one cutting edge (ts 5.0 feature), I put here first. can probably be used by isomer studio. later PR.